opennlp-tools/src/main/java/opennlp/tools/ml/model/Event.java

```diff
@@ -18,20 +18,23 @@
 package opennlp.tools.ml.model;
 
 
+import java.util.Objects;
+
 /**
  * The context of a decision point during training.  This includes
  * contextual predicates and an outcome.
  */
 public class Event {
-  private String outcome;
-  private String[] context;
-  private float[] values;
+  private final String outcome;
+  private final String[] context;
+  private final float[] values;
 
   public Event(String outcome, String[] context) {
     this(outcome,context,null);
   }
 
   public Event(String outcome, String[] context, float[] values) {
+    Objects.requireNonNull(context, "context must not be null");
```

Comment:
Probably nicer to write it like that:
this.context = Objects.requireNonNull(context, "context must not be null");

And the outcome should also be non-null.